src/components/PostDetails.tsx

```diff
@@ -1,106 +1,136 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { addComment, deleteComment, getComments } from '../api/comments';
+import { Comment } from '../types/Comment';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+type Props = {
+  post: Post;
+  isCommenting: boolean;
+  setIsCommenting: React.Dispatch<React.SetStateAction<boolean>>;
+};
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({
+  post,
+  isCommenting,
+  setIsCommenting,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loader, setLoader] = useState(false);
+  const [error, setError] = useState();
+  const [isLoading, setIsLoading] = useState(false);
 
-        <div className="block">
-          <Loader />
+  function loadComments() {
+    setLoader(true);
+    getComments(post.id)
+      .then(setComments)
+      .catch(e => {
+        setError(e);
+      })
+      .finally(() => setLoader(false));
+  }
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  function handleDeleteComment(commentId: number) {
+    deleteComment(commentId).catch(e => {
+      setComments(comments);
+      setError(e);
+      throw e;
```

Comment:
Fix previous comment 
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/70009205/1620dd0d-ee79-4ead-a849-52b427c4ae5f)